os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -79,20 +83,13 @@ static struct mm_ram_partition_s g_mempartitions[CONFIG_NUM_APPS];
 
 void mm_initialize_ram_partitions(void)
 {
-	int i = 0;
 	uint32_t start;
-	uint32_t size;
 
 	start = (uint32_t)__usram_segment_start__ + (uint32_t)__usram_segment_size__;
-	size = (REGION_END - start) / CONFIG_NUM_APPS;
-	for (i = 0; i < CONFIG_NUM_APPS; i++) {
-		g_mempartitions[i].start = ALIGN_UP(start);
-		g_mempartitions[i].size = size;
-		g_mempartitions[i].status = MM_PART_FREE;
-		start += size;
-
-		mvdbg("App RAM partition_%d start = 0x%x size = %u\n", i, g_mempartitions[i].start, g_mempartitions[i].size);
-	}
+	g_default_size = (REGION_END - start - sizeof(struct mm_heap_s)) / CONFIG_NUM_APPS;
```

Comment:
for each app size we need to remove sizeof(struct mm_heap_s)?

Reply:
I think there is a problem in the calculation here. Thank you for pointing this out. 
We will consider the mm_heap_s size and also some other overheads for each app and set the default partition size accordingly. I fixed it.